### PR TITLE
feat: add project & org auth flow (ENG-2628)

### PR DESCRIPTION
**Pull request by johnsca**
Depends on: https://github.com/stacklet/sinistral/pull/328

Description of usage and documentation added to the above PR: https://github.com/stacklet/sinistral/blob/5e2a8afc4def2c03fa22e3852e9fc19a60b59afe/docs/src/tutorial/ci-usage.md

### CODE REVIEW

**Review by tvansteenburgh — COMMENTED** [review]
Hey @johnsca, thanks for doing this work. Would you mind adding some more context to the PR description to help me understand what Project and Org auth are, how they differ from what we already had (SSO and user/pass I guess?), and when a user would pick one over the other (intended use cases)?

Just trying to understand the intended behavior while reading the code. Plus, that info will be useful for docs anyway.

**Comment by johnsca** [comment]
Yeah, of course. I was in a big rush putting this up; it definitely needs more polish, including better explanation / docs, and a bunch of test fixes and some added for the new auth flows which will also help. Changing this to a draft for now.

**Review by tvansteenburgh — APPROVED** [review]
Some nice refactors in here, thanks @johnsca. LGTM once those last two tests are fixed/passing. I'll take a look at that next.

**Review by thisisshi — APPROVED** [review]

File: stacklet/client/sinistral/config.py [hunk 1]
```diff
@@ -4,73 +4,110 @@
...
 
 
 class StackletConfig:
+    CONFIG_FILE = "config.json"
+
     schema = {
         "type": "object",
         "properties": {
-            "api": {"type": "string"},
+            "api_url": {"type": "string"},
```
[the remark this answers is not in the document]
Comment (johnsca) [reply stacklet/client/sinistral/config.py:20]: There is [explicit logic to ensure backward compatibility](https://github.com/stacklet/sinistral-cli/pull/28/files#diff-6da989c4320867303696689d43ed7d8b203324a67d423280f6f5038100e9fe4bR80-R87), as well as test coverage [here](https://github.com/stacklet/sinistral-cli/pull/28/files#diff-4e8715c7a425ee52e74b7df4d34efd32e8c92f3e60bd51bc2e1ad5943b82032eR53-R73) and [here](https://github.com/stacklet/sinistral-cli/pull/28/files#diff-da24b355349b53e4c8047034767d008e4c2ce9ea6c8933a3acd50d0828f00528R30-R33) to make sure it's working.

However, this does make me realize that I didn't change how the hosted config file is generated and they still use the old format. Though I'm actually inclined to leave them using the old format for now, with a note to update once the CLI change is out for a while.

**Review by thisisshi — APPROVED** [review]

**Pull request merged**